### ISSUE TITLE
Initialise a CTFd instance from juice-shop-ctf output

### ISSUE THREAD

**Issue by PsypherPunk** [PsypherPunk/terraform-provider-ctfd#2]
As per [`tf-aws-service-owasp-juice-shop`](https://github.com/PsypherPunk/tf-aws-service-owasp-juice-shop), we can now generate a CTFd config. from [`juice-shop-ctf`](https://github.com/juice-shop/juice-shop-ctf) via Terraform.

Assuming that we now have a generated config. file available, 

The process is (somewhat unfortunately):

- Set up the CTFd instance (name, description, admin. details, etc.);
- Log in as the newly-created admin. user;
- Upload the aforementioned file;
- Set up the CTFd instance (*again*.)

For reference, this was the Bash function used to set up CTFd earlier:

```bash
function setup() {
    nonce=$(curl "${ctfd_url}/setup" \
        --silent \
        --cookie "{cookie_jar}" \
        --cookie-jar "${cookie_jar}" | grep csrfNonce | cut -d '"' -f2)
    curl "${ctfd_url}/setup" \
        --silent \
        --request POST \
        --cookie "${cookie_jar}" \
        --cookie-jar "${cookie_jar}" \
        --data "nonce=${nonce}" \
        --data "ctf_name=${ctf_name}" \
        --data "ctf_description=${ctf_description}" \
        --data "name=${admin_name}" \
        --data "user_mode=${user_mode}" \
        --data-urlencode "email=${admin_email}" \
        --data-urlencode "password=${admin_password}" 2>&1 1> /dev/null
}
```

And this for uploading the config. file:

```bash
echo "Uploading Juice Shop CTF challenges…"
nonce=$(curl "${ctfd_url}/admin/config" \
    --silent \
    --cookie-jar "${cookie_jar}" \
    --cookie "${cookie_jar}" | grep csrfNonce | cut -d '"' -f2)
curl "${ctfd_url}/admin/import" \
    --dump-header - \
    --silent \
    --request POST \
    --cookie "${cookie_jar}" \
    --cookie-jar "${cookie_jar}" \
    --form "nonce=${nonce}" \
    --form "backup=@${ctfd_config}" 2>&1 > /dev/null
```

**Comment by PsypherPunk** [comment]
Extend the existing `provider` to include:

- `name`
- `description`
- `configuration_path`

Currently in `api/auth` we check that we're already setup (see `CheckSetup()`.) Should we conditionally check that, if `configuration_path` is set, the opposite is true? And then proceed with setting it up?

Or should there be a separate resource to hold the configuration?

```hcl
resource "ctfd_setup" "setup" {
  name = "Sample CTFd"
  description = "Nothing to see here."
  configuration_path = "/tmp/filename.zip"
}
```

There would have to be a `depends_on` for the teams/users to avoid the chance we'll create teams/users in an instance we then nuke but that could work.

**Comment by PsypherPunk** [comment]
Implemented in 386f66d6519e92c388434c9356df07b60eee361c.
